exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitExtension.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.testkit;
+
+import com.exonum.binding.util.LibraryLoader;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+public class TestKitExtension implements ParameterResolver, BeforeEachCallback, AfterEachCallback {
+
+  private static final String KEY = "ResourceKey";
+
+  static {
+    LibraryLoader.load();
+  }
+
+  private final TestKit.Builder testKitBuilder;
+
+  public TestKitExtension(TestKit.Builder testKitBuilder) {
+    this.testKitBuilder = testKitBuilder;
+  }
+
+  @Override
+  public void beforeEach(ExtensionContext extensionContext) {
```

Comment:
Why beforeEach is implemented, creating the Testkit even if it is not requested? I think it must be instantiated when requested (= resolved).